ngraph/test/util/engine/ie_cpu_engine.hpp

```diff
@@ -0,0 +1,260 @@
+//*****************************************************************************
+// Copyright 2017-2020 Intel Corporation
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//*****************************************************************************
+
+#pragma once
+
+#include <ie_core.hpp>
+#include "ngraph/function.hpp"
+#include "util/all_close.hpp"
+#include "util/all_close_f.hpp"
+#include "util/engine/engine_traits.hpp"
+
+namespace ngraph
+{
+    namespace test
+    {
+        class IE_CPU_Engine
```

Comment:
IMHO we need some interface class here, otherwise how would you request the user to provide correct API? Moreover If I would like to write another similar Engine class I don't know what interface it should provide until I look into the TestCase code where it would be used.